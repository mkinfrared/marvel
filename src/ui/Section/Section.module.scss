@import "styles/colors";

.Section {
  margin-top: 40px;

  > h4 {
    font-size: 24px;
    font-weight: 600;
    margin-bottom: 20px;
    text-align: left;
  }

  > hr {
    border-color: $primary-dark;
    margin: 20px 0;
  }

  .list {
    display: grid;
    grid-gap: 20px;
    grid-template-columns: repeat(auto-fill, minmax(200px, 200px));
    justify-content: space-evenly;
  }
}
