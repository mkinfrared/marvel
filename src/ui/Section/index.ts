export { default } from "ui/Section/Section";
