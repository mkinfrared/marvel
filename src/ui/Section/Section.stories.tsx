import { Meta, Story } from "@storybook/react/types-6-0";
import React from "react";

import Card from "ui/Card";

import { Section } from "./Section";
import { SectionProps } from "./Section.type";

export default {
  title: "UI/Section",
  component: Section
} as Meta;

const Template: Story<SectionProps> = (args) => (
  <Section {...args}>
    <Card
      title="Dr. Strange"
      imageSrc="http://i.annihil.us/u/prod/marvel/i/mg/6/f0/5239b5e7d7f70.jpg"
    />
    <Card
      title="Dr. Strange"
      imageSrc="http://i.annihil.us/u/prod/marvel/i/mg/6/f0/5239b5e7d7f70.jpg"
    />
    <Card
      title="Dr. Strange"
      imageSrc="http://i.annihil.us/u/prod/marvel/i/mg/6/f0/5239b5e7d7f70.jpg"
    />
    <Card
      title="Dr. Strange"
      imageSrc="http://i.annihil.us/u/prod/marvel/i/mg/6/f0/5239b5e7d7f70.jpg"
    />
    <Card
      title="Dr. Strange"
      imageSrc="http://i.annihil.us/u/prod/marvel/i/mg/6/f0/5239b5e7d7f70.jpg"
    />
    <Card
      title="Dr. Strange"
      imageSrc="http://i.annihil.us/u/prod/marvel/i/mg/6/f0/5239b5e7d7f70.jpg"
    />
    <Card
      title="Dr. Strange"
      imageSrc="http://i.annihil.us/u/prod/marvel/i/mg/6/f0/5239b5e7d7f70.jpg"
    />
    <Card
      title="Dr. Strange"
      imageSrc="http://i.annihil.us/u/prod/marvel/i/mg/6/f0/5239b5e7d7f70.jpg"
    />
    <Card
      title="Dr. Strange"
      imageSrc="http://i.annihil.us/u/prod/marvel/i/mg/6/f0/5239b5e7d7f70.jpg"
    />
    <Card
      title="Dr. Strange"
      imageSrc="http://i.annihil.us/u/prod/marvel/i/mg/6/f0/5239b5e7d7f70.jpg"
    />
  </Section>
);

const Primary = Template.bind({});

Primary.args = {
  title: "Dr. Strange images"
};

export { Primary };
