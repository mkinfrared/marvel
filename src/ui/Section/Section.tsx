import React from "react";

import css from "./Section.module.scss";
import { SectionProps } from "./Section.type";

const Section = ({ children, title }: SectionProps) => {
  return (
    <section className={css.Section}>
      <h4>{title}</h4>
      <hr />
      <div className={css.list}>{children}</div>
    </section>
  );
};

export default React.memo(Section);
