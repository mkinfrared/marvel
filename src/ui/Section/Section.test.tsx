import { render } from "@testing-library/react";
import React from "react";

import Section from "./Section";

describe("<Section />", () => {
  const Component = Section;
  const title = "marklar";
  const children = "foobar";
  const props = { title, children };

  it("should be defined", () => {
    const { container } = render(<Component {...props} />);

    expect(container).toBeDefined();
  });

  it("match the snapshot", () => {
    const { container } = render(<Component {...props} />);

    expect(container).toMatchSnapshot();
  });
});
