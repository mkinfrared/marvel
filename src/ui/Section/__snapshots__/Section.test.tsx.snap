// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Section /> match the snapshot 1`] = `
<div>
  <section
    class="Section"
  >
    <h4>
      marklar
    </h4>
    <hr />
    <div
      class="list"
    >
      foobar
    </div>
  </section>
</div>
`;
